napari/utils/translations.py

```diff
@@ -0,0 +1,456 @@
+"""
+Localization utilities to find available language packs and packages with
+localization data.
+"""
+
+import gettext
+import importlib
+import json
+import os
+import subprocess
+import sys
+import traceback
+from pathlib import Path
+
+import pkg_resources
+
+# Entry points
+NAPARI_LANGUAGEPACK_ENTRY = "napari.languagepack"
+
+# Constants
+DEFAULT_LOCALE = "en"
+LOCALE_DIR = "locale"
+
+
+def _main():
+    """
+    Run functions in this file in a subprocess and prints to stdout the results.
+    """
+    data = {}
+    message = ""
+    if len(sys.argv) == 2:
+        func_name = sys.argv[-1]
+        func = globals().get(func_name, None)
+
+        if func:
+            try:
+                data, message = func()
+            except Exception:
+                message = traceback.format_exc()
+    else:
+        message = "Invalid number of arguments!"
+
+    sys.stdout.write(json.dumps({"data": data, "message": message}))
+
+
+def _get_installed_language_pack_locales() -> dict:
+    """
+    Get available installed language pack locales.
+
+    Returns
+    -------
+    dict
+        A dict with all the languages found.
+
+    Notes
+    -----
+    This functions are meant to be called via a subprocess to guarantee the
+    results represent the most up-to-date entry point information, which
+    is defined on interpreter startup.
+    """
+    data = {}
+    messages = []
+    for entry_point in pkg_resources.iter_entry_points(
+        NAPARI_LANGUAGEPACK_ENTRY
+    ):
+        try:
+            data[entry_point.name] = os.path.basename(
+                os.path.dirname(entry_point.load().__file__)
+            )
+        except Exception:
+            messages.append(traceback.format_exc())
```

Comment:
```suggestion
    if sys.version_info >= (3, 8):
        from importlib import metadata as importlib_metadata
    else:
        import importlib_metadata

    data = {}
    messages = []
    for dist in importlib_metadata.distributions():
        for ep in dist.entry_points:
            if ep.group == NAPARI_LANGUAGEPACK_ENTRY:
                try:
                    data[ep.name] = ep.value
                except Exception:
                    messages.append(traceback.format_exc())
```
we can avoid `pkg_resources` all together, which can add a lot of time to our imports depending on the user's environment (see https://github.com/napari/napari/pull/2121 and discussion in [this readme](https://github.com/takluyver/entrypoints)), by using the `importlib.metadata` from the standard library.  This is the entry point discovery mechanism that we use in napari-plugin-engine.

In fact, a lot of this entry-point discovery code is duplicating logic already contained in napari-plugin-engine...  I'll look into seeing how you might be able to use that

Reply:
you can also use:
```python
from napari_plugin_engine.manager import iter_available_plugins

data = {k: v for (k, v, _) in iter_available_plugins(NAPARI_LANGUAGEPACK_ENTRY)}
```